test/runtests.jl

```diff
@@ -477,5 +477,37 @@ using Compat
             @test_broken isready(bottom_dc3) == true
             @test_broken fetch(bottom_dc3) == 14
         end
+
+        @testset "DeferredFuture serialization as part of another object" begin
```

Comment:
Yeah, these tests look a lot cleaner now. It's also nice that it clearly documents the old failure condition outside of dispatcher.